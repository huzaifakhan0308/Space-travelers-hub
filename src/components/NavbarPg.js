import { NavLink } from 'react-router-dom';
import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import logo from '../Assets/planet.png';

const NavbarPg = () => (
  <Navbar
    collapseOnSelect
    expand="lg"
    bg="dark"
    variant="dark"
    className=" border-bottom  "
  >
    <Container>
      <NavLink to="/" className="fw-bolder fs-3 text-white">
        <img
          alt=""
          src={logo}
          width="50"
          height="40"
          className="d-inline-block align-top"
        />
        {' '}
        Space Travelers&lsquo; Hub
      </NavLink>
      <Navbar.Toggle aria-controls="responsive-navbar-nav" />
      <Navbar.Collapse id="responsive-navbar-nav" className="fs-5 ">
        <Nav className="me-auto  flex-grow-1 pe-3 justify-content-end ">
          <NavLink
            to="rockets"
            className={({ isActive }) => (isActive
              ? 'linkNvb text-warning fw-bold border-top border-bottom border-warning'
              : ' text-white linkNvb')}
          >
            Rockets
          </NavLink>
          <NavLink
            to="missions"
            end
            className={({ isActive }) => (isActive
              ? 'linkNvb text-warning fw-bold border-top border-bottom border-warning'
              : ' text-white linkNvb')}
          >
            Missions
          </NavLink>
          <div className="vr bg-white mx-3" />
          <NavLink
            to="my-profile"
            className={({ isActive }) => (isActive
              ? 'linkNvb text-warning fw-bold border-top border-bottom border-warning'
              : ' text-white linkNvb')}
          >
            My Profile
          </NavLink>
        </Nav>
      </Navbar.Collapse>
    </Container>
  </Navbar>
);

export default NavbarPg;
