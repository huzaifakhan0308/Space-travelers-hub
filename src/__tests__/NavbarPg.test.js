import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import NavbarPg from '../components/NavbarPg';

it('renders correctly', () => {
  const navigation = renderer
    .create(<Router><NavbarPg /></Router>)
    .toJSON();
  expect(navigation).toMatchSnapshot();
});
