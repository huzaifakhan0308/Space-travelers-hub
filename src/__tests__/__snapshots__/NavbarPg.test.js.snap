// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className=" border-bottom   navbar navbar-expand-lg navbar-dark bg-dark"
>
  <div
    className="container"
  >
    <a
      aria-current="page"
      className="fw-bolder fs-3 text-white active"
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className="d-inline-block align-top"
        height="40"
        src="planet.png"
        width="50"
      />
       
      Space Travelersâ€˜ Hub
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="fs-5  navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="me-auto  flex-grow-1 pe-3 justify-content-end  navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className=" text-white linkNvb"
          href="/rockets"
          onClick={[Function]}
        >
          Rockets
        </a>
        <a
          className=" text-white linkNvb"
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
        <div
          className="vr bg-white mx-3"
        />
        <a
          className=" text-white linkNvb"
          href="/my-profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </div>
    </div>
  </div>
</nav>
`;
